packages/aws-cdk-lib/aws-cognito/lib/managed-login-branding.ts

```diff
@@ -0,0 +1,432 @@
+  IDP_BUTTON_ICON = 'IDP_BUTTON_ICON',
+}
+
+/**
+ * The display-mode target of the asset: light, dark, or dynamic (browser-adaptive).
+ */
+export enum ColorMode {
+  /**
+   * Light mode
+   */
+  LIGHT = 'LIGHT',
+
+  /**
+   * Dark mode
+   */
+  DARK = 'DARK',
+
+  /**
+   * Browser-adaptive mode (displays in all contexts)
+   */
+  DYNAMIC = 'DYNAMIC',
+}
+
+/**
+ * The file type of the image file.
+ */
+export enum AssetExtension {
+  /**
+   * ICO file type
+   */
+  ICO = 'ICO',
+
+  /**
+   * JPEG file type
+   */
+  JPEG = 'JPEG',
+
+  /**
+   * PNG file type
+   */
+  PNG = 'PNG',
+
+  /**
+   * SVG file type
+   */
+  SVG = 'SVG',
+
+  /**
+   * WEBP file type
+   */
+  WEBP = 'WEBP',
+}
+
+/**
+ * An asset for managed login branding
+ */
+export interface BrandingAsset {
+  /**
+   * The category that the image corresponds to in your managed login configuration.
+   * Managed login has asset categories for different types of logos, backgrounds, and icons.
+   */
+  readonly category: AssetCategory;
+
+  /**
+   * The display-mode target of the asset: light, dark, or browser-adaptive.
+   * For example, Amazon Cognito displays a dark-mode image only when the browser or application
+   * is in dark mode, but displays a browser-adaptive file in all contexts.
+   */
+  readonly colorMode: ColorMode;
+
+  /**
+   * The file type of the image file.
+   */
+  readonly extension: AssetExtension;
+
+  /**
+   * The image file, in Base64-encoded binary.
+   *
+   * @default - No bytes are provided
+   */
+  readonly bytes?: string;
+
+  /**
+   * The ID of the asset.
+   *
+   * @default - No resource ID is provided
+   */
+  readonly resourceId?: string;
+}
+
+/**
+ * Represents a managed login branding for an Amazon Cognito User Pool.
+ */
+export interface IManagedLoginBranding extends IResource {
+  /**
+   * The ID of the managed login branding style.
+   * @attribute
+   */
+  readonly managedLoginBrandingId: string;
+}
+
+/**
+ * Properties for creating a ManagedLoginBranding
+ */
+export interface ManagedLoginBrandingProps {
+  /**
+   * The user pool ID where the branding style is assigned.
+   */
+  readonly userPoolId: string;
+
+  /**
+   * The app client ID that you want to assign the branding style to.
+   *
+   * Each style is linked to an app client until it is deleted.
+   *
+   * @default - No specific client ID is assigned
+   */
+  readonly clientId?: string;
+
+  /**
+   * Name for the managed login branding.
+   *
+   * @default - A name is automatically generated.
+   */
+  readonly managedLoginBrandingName?: string;
+
+  /**
+   * An array of image files that you want to apply to roles like backgrounds, logos, and icons.
+   *
+   * Each object must also indicate whether it is for dark mode, light mode, or browser-adaptive mode.
+   *
+   * @default - No custom assets are provided
+   */
+  readonly assets?: BrandingAsset[];
+
+  // TODO: create a detailed type for settings as there isn't a json schema published yet.
```

Comment:
unfortunately this cannot be fully typed yet as there is no json schema published from cognito a